node/libs/roles/src/validator/messages/leader_commit.rs

```diff
@@ -54,6 +54,26 @@ pub enum CommitQCVerifyError {
     BadSignature(#[source] anyhow::Error),
 }
 
+/// Error returned by `CommitQC::add()`.
+#[derive(thiserror::Error, Debug)]
+pub enum CommitQCAddError {
```

Comment:
kudos for adding strong error types here. How about adding a small test for those methods while you are at it?